badge_server/test_badges.py

```diff
@@ -1053,6 +1053,121 @@ def test_github_py2py3_py2_installation_failure_fresh_nodeps(self):
         self.assertTargetResponse(package_name, expected_pair_result)
 
 
+class TestPairIncompatibility(BadgeTestCase):
+    """Test for cases where the badge image displays 'pair incompatible.'"""
+
+    def assertImageResponsePyPI(self, package_name):
+        """Assert that the badge image response is correct for a PyPI package."""
+        BadgeTestCase._assertImageResponsePyPI(
+            self, package_name, main.BadgeStatus.PAIR_INCOMPATIBLE)
+
+    def assertImageResponseGithub(self, package_name):
+        """Assert that the badge image response is correct for a github package."""
+        BadgeTestCase._assertImageResponseGithub(
+            self, package_name, main.BadgeStatus.PAIR_INCOMPATIBLE)
+
+    def assertTargetResponse(self, package_name, expected_pair_result):
+        json_response = self.get_target_json(package_name)
+        self.assertEqual(json_response['package_name'], package_name)
+        self.assertBadgeStatusToColor(json_response['badge_status_to_color'])
+
+        # self compatibility result check
+        expected_status = main.BadgeStatus.SUCCESS
+        self.assertEqual(
+            json_response['self_compat_res'],
+            utils._build_default_result(expected_status))
```

Comment:
Do we do this in other places? You really want to minimize logic in tests. Otherwise you have to write a test to validate the tests.

Reply:
We have been doing this in other places

Reply:
I was trying to avoid repetitive declarations to improve readability and ease of maintenance, but I don't have a problem with axing these.